src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -85,6 +78,9 @@ public bool TryFetch(object obj, out T value, bool skipObjNullCheck = false)
         }
 
         // see https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
+#if NET6_0_OR_GREATER
```

Comment:
As discussed in other places, I think it would be cleaner to use the attribute always, regardless of TFM, but it requires solving the other problems with adding the internal definition of the attributes. I just had to put it here :-)